src/nodes/VectorNode.ts

```diff
@@ -56,6 +56,24 @@ export class VectorNode extends ArgumentNode implements ArrayLike<VectorElementN
         return ans
     }
 
+    getChunk() {
+        if (this.length !== 2 && this.length !== 3) {
+            return undefined
+        }
+        const vexX = this[0]
+        const vexZ = this.length === 3 ? this[2] : this[1]
+        // TODO: support VectorElementType.Relative
+        if (vexX.type != VectorElementType.Absolute || vexZ.type != VectorElementType.Absolute) {
+            return undefined
+        }
+        return Object.assign(new VectorNode(), {
+            length: 2,
+            0: new VectorElementNode(VectorElementType.Absolute, Math.floor(vexX.valueOf() / 16), ''),
+            1: new VectorElementNode(VectorElementType.Absolute, Math.floor(vexZ.valueOf() / 16), ''),
+            [NodeRange]: this[NodeRange]
+        })
```

Comment:
This is quite tricky. I personally prefer using `push` method provided by `VectorNode` directly, in case the internal structure of `VectorNode` changed in the future.

```suggestion
        const ans = new VectorNode()
        ans.push(
            new VectorElementNode(VectorElementType.Absolute, Math.floor(vexX.valueOf() / 16), ''),
            new VectorElementNode(VectorElementType.Absolute, Math.floor(vexZ.valueOf() / 16), '')
        })
        ans[NodeRange] = this[NodeRange]
        return ans
```